internal/upload/uploader/uploader.go

```diff
@@ -428,6 +445,25 @@ func (u *Uploader) tryRefresh(ctx context.Context) bool {
 	return true
 }
 
+func (req *SectorUploadReq) Finish(hk types.PublicKey, fcid types.FileContractID, err error) {
+	req.mu.Lock()
+	defer req.mu.Unlock()
+	req.once.Do(func() {
+		req.ResponseChan <- SectorUploadResp{
+			FCID: fcid,
+			HK:   hk,
+			Err:  err,
+			Req:  req,
+		}
+	})
+}
+
+func (req *SectorUploadReq) Reset() {
+	req.mu.Lock()
+	defer req.mu.Unlock()
+	req.once = sync.Once{}
```

Comment:
this is embarrassing

Reply:
Yeah this version was never meant to get merged, just meant to highlight the issue and to test it on erravimus. Added separate type 👍